### ISSUE TITLE
Update version of Alpine to build EVE

### ISSUE THREAD

**Issue by famleebob** [lf-edge/eve#3994]
Currently, EVE is built with Alpine 3.16.  Version 3.16 is now out of support.  Need to update to supported version.  Suggest the recently released 3.20.

**Comment by famleebob** [comment]
Alpine creates two stable releases each year (see [releases page](https://alpinelinux.org/releases/)).  The main repository is supported for two years, but the community repository is supported until the next stable release (should be six months).  The ```edge``` repository is "rolling" and thus is subject to constant change.   There are three releases that are candidates for this update
| Release | Rel Date | EOL | kernel-lts | gcc | go | rust |
|---|---|---|---|---|---|---|
| 3.22 | 05/22/2025 | 5/1/2027 | 6.12.41 | 14.2.0 | 1.24.6 | 1.87.0 |
| 3.21 | 12/05/2024 | 11/01/2026 | 6.12.41 | 14.2.0 | 1.23.9 | 1.83.0 |
| 3.20 | 05/22/2024 | 04/01/2026 | 6.6.101 | 13.2.1 | 1.22.10 | 1.78.0 |

**NOTE**: Package versions from [Alpine Linux Package Index](https://pkgs.alpinelinux.org/packages)

There is a policy, that EVE utilizes the ```linux-lts``` that has been in use for one year.  That indicates 3.20 or 3.21 (3.21 passes one year December 5<sup>th</sup>).  3.22 will have roughly five months "soak-time", but would get us the version of ```go``` desired.  We currently pull ```go``` from the ```edge``` roulette wheel (version from ```edge``` being a gamble).  I'd rather use the download from the [go download web page](https://go.dev/dl/) install it in the base distro container (alpine), and get the exact version desired, and puts version changes squarely in our hands.

In this case to define the ```distro``` container, that it provides a package cache, common base root file system, and capable of generating an EVE service container.

Management has expressed a desire for this in October to complement the new kernel release.

My suggestion is to take either 3.21 or 3.22.  With the edge to 3.22, even with the shorter soak time.  The ```go``` resolution without pulling from ```eddge``` pushes me toward it.  Just need to pick the version and get started.

**Comment by rene** [comment]
I agree with you @famleebob , my vote is to go with Alpine 3.22. Install Go from Alpine or direct download it's another discussion and it can be done separated from this issue.

**Comment by deitch** [comment]
Agreed as well, we should go for 3.22.

The challenge is the packages. Updating from 3.16 to 3.22 will update a _lot_ of package versions. Some will even have been removed or split into others (I have had this issue in the past). To take the "simplest" 😬 example, the [pkg list for pillar](https://github.com/lf-edge/eve/blob/master/pkg/pillar/Dockerfile#L7-L8) is:

```
git gcc linux-headers libc-dev make linux-pam-dev m4 \
 findutils go util-linux make patch \
 libintl libuuid libtirpc libblkid libcrypto1.1 zlib tar
```

Those certainly will have changed a lot, and likely something will break. We would need to rebuild and republish literally every single package. That is likely why we haven't done it until now. We got reproducible builds with controlled package (not the usual alpine uncertainty) at the price of needing to invest time to update (and therefore pretty much not doing it).

The upside is that the rebuild of all of the packages will find most problems right away, and the eden tests any lingering ones, although I do not expect any at the later stage (but you never know).

I suspect this will require someone dedicating at least several days to it.

**Comment by rucoder** [comment]
@famleebob I vote for 3.22 too!

**Comment by rene** [comment]
Closing since Alpine was successfully upgraded.
